controllers/template_sync.go

```diff
@@ -118,19 +119,29 @@ func (r *PolicyReconciler) Reconcile(ctx context.Context, request reconcile.Requ
 		return reconcile.Result{}, nil
 	}
 
-	// In certain situations, we should requeue without an error, *after* other logic.
-	// Use this in non-error returns to correctly handle those situations.
-	result := reconcile.Result{}
+	// Do not exit early from the loop - store an error to return later and `continue`. Be careful
+	// not to overwrite the error in a way that it becomes nil, which would prevent a requeue.
+	// As a quirk of the error handling, only the last occurring error is "returned" by Reconcile.
+	var resultError error
 
 	// PolicyTemplates is not empty
 	// loop through policy templates
-	for _, policyT := range instance.Spec.PolicyTemplates {
+	for tIndex, policyT := range instance.Spec.PolicyTemplates {
 		object, gvk, err := unstructured.UnstructuredJSONScheme.Decode(policyT.ObjectDefinition.Raw, nil, nil)
+
+		tName := "*" // placeholder for the template name - status sync might need to specially handle this
```

Comment:
When I was thinking status-sync would be updated to handle some additional cases, I was planning on implementing something with this. But it got put aside. So right now, it is not used except in error messages.

`*` is probably not the best value, but I'm not positive what to use instead. Since the value is used in logs and error messages, maybe it should make it more obvious the the template name could not be determined, or that no name was provided in the template? I think an empty string could just add confusion later.

What about `[unknown template index <n>]` (or similar, but more succinct), so that it identifies where the problem is coming from?